modules/json/include/inviwo/json/jsonutils.h

```diff
@@ -0,0 +1,57 @@
+/*********************************************************************************
+ *
+ * Inviwo - Interactive Visualization Workshop
+ *
+ * Copyright (c) 2019 Inviwo Foundation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+#pragma once
+
+#include <inviwo/json/jsonmoduledefine.h>
+#include <inviwo/core/common/inviwo.h>
+#include <modules/plotting/datastructures/dataframe.h>
+#include <nlohmann/json.hpp>
+
+namespace inviwo {
+
+template <typename T>
+std::string dataframeToJsonString(std::vector<std::shared_ptr<const T>> dataFrame) {
+
+    using json = nlohmann::json;
+    json root;  // = json::array();
+
+    for (auto row = 0; row < dataFrame->getNumberOfRows(); ++row) {
+        json node = json::object();
+        auto items = dataFrame->getDataItem(row, true);
+        int i = 1;
+        for (auto col = ++items.begin(); col != items.end(); ++col) {
+            node[dataFrame->getHeader(i++)] = (*col)->toString();
+        }
+        root.emplace_back(node);
+    }
+
+    return root.dump();
```

Comment:
Should return json.
Rename function to dataFrameToJson

Reply:
Then maybe also create a json to string conversion function? Or else the webbrowser and other processors that only need the string will also need to be able to handle json objects.

Reply:
If a string is desired one can just call dump() on the json object